scripts/build/build/targets.py

```diff
@@ -119,10 +143,54 @@ def HostTargets():
         app_targets.append(target.Extend('thermostat', app=HostApp.THERMOSTAT))
         app_targets.append(target.Extend('minmdns', app=HostApp.MIN_MDNS))
 
+    # Possible build variantes. Note that number of potential
+    # builds is exponential here
+    variants = [
+        HostBuildVariant(name="ipv6only", enable_ipv4=False),
+        HostBuildVariant(name="no-ble", enable_ble=False),
+        HostBuildVariant(name="tsan", conflicts=['asan'], use_tsan=True),
+        HostBuildVariant(name="asan", conflicts=['tsan'], use_asan=True),
+        HostBuildVariant(name="same-event-loop",
+                         validator=AcceptNameWithSubstring('-chip-tool'), separate_event_loop=False),
+    ]
+
+    glob_whitelist = set(['ipv6only'])
+
     for target in app_targets:
         yield target
-        if ('rpc-console' not in target.name):
-            yield target.Extend('ipv6only', enable_ipv4=False)
+
+        if 'rpc-console' in target.name:
+            # rpc console  has only one build variant right now
+            continue
+
+        # skip variants that do not work for  this target
+        ok_variants = [v for v in variants if v.validator.Accept(target.name)]
+
+        # Build every possible variant
+        for variant_count in range(1, len(ok_variants) + 1):
+            for subgroup in combinations(ok_variants, variant_count):
+                # find if a subgroup contains a conflict
+                conflict = False
+                for a, b in combinations(subgroup, 2):
+                    if (a.name in b.conflicts) or (b.name in a.conflicts):
+                        conflict = True
+                        break
+
+                if conflict:
+                    continue
```

Comment:
I'm not sure if this can use [for/else clause](https://book.pythontips.com/en/latest/for_-_else.html#else-clause)
```suggestion
                for a, b in combinations(subgroup, 2):
                    if (a.name in b.conflicts) or (b.name in a.conflicts):
                        conflict = True
                        break
                else:
                    continue
```

Reply:
I find this variant harder to read. Moved to a separate function.